src/components/OrderBookChart/index.tsx

```diff
@@ -131,6 +130,19 @@ export const Chart: React.FC<ChartProps> = props => {
       networkId,
     })
 
+    let firstLoad = true
```

Comment:
can't this be achieved the same essentially with `onComponentMount` via

`useEffect(() => { ... }, [])`?

this is all this is doing no?

Reply:
Unless there's a way I'm not seeing, I don't think so.

The reasoning is:

I only want to change the zoom, when the data is loaded for the first time, to center the graph.
The zoom coordinates are calculated every time the data is loaded, in case the user wants to recenter it after playing around.
If I update the zoom in another `useEffect`, I need to depend on the zoom coordinates.
If I depend on the coordinates, the `useEffect` will trigger every time it changes, which is every time the data is loaded, which is not what I want.
Thus, the `firstLoad` flag tracks when it has been loaded already and prevents resetting the zoom.

@W3stside do you see another way of doing that?